backend/core/src/applications/applications.service.ts

```diff
@@ -51,6 +58,13 @@ export class ApplicationsService {
 
   async submit(applicationCreateDto: ApplicationCreateDto) {
     applicationCreateDto.submissionDate = new Date()
+    const listing = await this.listingsService.findOne(applicationCreateDto.listing.id)
+    if (
+      listing.applicationDueDate &&
+      applicationCreateDto.submissionDate > listing.applicationDueDate
+    ) {
+      throw new BadRequestException("Listing is not open for application submission.")
+    }
```

Comment:
Can you update listingService findOne to take an optional `select: string[]` arg, and then pass in `applicationCreateDto.listing.id, ["applicationDueDate"]` to it. findOne will need a little bit of restructuring, but I don't think we need to fetch the entire record if we only need one field. I'm sure there are other places that could benefit from this too.

Reply:
@seanmalbert Do you think `list` endpoint should also have such functionality? Makes me wonder if this should be a separate refactoring PR.

Reply:
@pbn4 ,
I think that's a good idea. I created #1435, to that end, so we're good to merge this.